src/elf.rs

```diff
@@ -1136,10 +1140,12 @@ impl<V: Verifier, C: ContextObject> Executable<V, C> {
                         // Else it's a syscall
                         let hash = *syscall_cache
                             .entry(symbol.st_name())
-                            .or_insert_with(|| ebpf::hash_symbol_name(name.as_bytes()));
+                            .or_insert_with(|| ebpf::hash_symbol_name(name));
                         if config.reject_broken_elfs && loader.lookup_function(hash).is_none() {
                             return Err(ElfError::UnresolvedSymbol(
-                                name.to_string(),
+                                std::str::from_utf8(name)
+                                    .unwrap_or("UTF-8 error")
```

Comment:
Nit:

Couldn't this be `String::from_utf8_lossy(name).to_owned()`? The replacement
character is deterministic.

If for abundance of caution you want to use a fixed string in case the utf8
replacement character changes 😋 I think we should make our own
`from_utf8_lossy(bytes) -> String` that unwraps with `UTF-8 error` instead of
repeating it at every call site